### PR TITLE
[arci-ros] Add test for some `ROS` client.

### PR DESCRIPTION
**Pull request by kaaatsu32329**
The purpose is to increase coverage about `arci-ros`. This time, I implement tests for `RosRobotClient` and `RosSpeakClient`, and `RosLocalizationClient`. But a test of `RosLocalizationClient` does not cover all file lines. `~Builder` and `~Config` is not covered.

### CODE REVIEW

File: arci-ros/tests/test_robot_client.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    String::from("joint1"),
+                    String::from("joint2"),
+                ],
+                position: vec![0f64, 1f64, 2f64],
+                velocity: vec![3f64, 4f64, 5f64],
+                effort: vec![6f64, 7f64, 8f64],
+            })
+            .unwrap();
+
+        while rosrust::is_ok() && joint_trajectory_subscriber.publisher_count() == 0 {
```
[the remark this answers is not in the document]
Comment (kaaatsu32329) [reply arci-ros/tests/test_robot_client.rs:53]: Yes, but publisher waiting in L53 and publisher in L39 are different.

[the remark this answers is not in the document]
Comment (OTL) [reply arci-ros/tests/test_robot_client.rs:54]: I see, what you mean.

**Review by OTL — APPROVED** [review]

File: arci-ros/tests/test_robot_client.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#[cfg(target_os = "linux")]
```
Comment (OTL) [inline arci-ros/tests/test_robot_client.rs:1]: I've merged, but this means that this files' L2 is included.
It cases error in other OSs.
#![cfg] will work for whole file.

Comment (OTL) [reply arci-ros/tests/test_robot_client.rs:1]: And it can be run on MacOS not only linux.

**Pull request merged**